python/amici/parameter_mapping.py

```diff
@@ -0,0 +1,349 @@
+        Mapping for fixed preequilibration parameters.
+    :param scale_map_preeq_fix:
+        Scales for fixed preequilibration parameters.
+    :param map_sim_fix:
+        Mapping for fixed simulation parameters.
+    :param scale_map_sim_fix:
+        Scales for fixed simulation parameters.
+    """
+
+    def __init__(
+            self,
+            map_sim_var: SingleParameterMapping = None,
+            scale_map_sim_var: SingleScaleMapping = None,
+            map_preeq_fix: SingleParameterMapping = None,
+            scale_map_preeq_fix: SingleScaleMapping = None,
+            map_sim_fix: SingleParameterMapping = None,
+            scale_map_sim_fix: SingleScaleMapping = None,
+    ):
+        if map_sim_var is None:
+            map_sim_var = {}
+        self.map_sim_var = map_sim_var
+
+        if scale_map_sim_var is None:
+            scale_map_sim_var = {key: LIN for key in map_sim_var}
+        self.scale_map_sim_var = scale_map_sim_var
+
+        if map_preeq_fix is None:
+            map_preeq_fix = {}
+        self.map_preeq_fix = map_preeq_fix
+
+        if scale_map_preeq_fix is None:
+            scale_map_preeq_fix = {key: LIN for key in map_preeq_fix}
+        self.scale_map_preeq_fix = scale_map_preeq_fix
+
+        if map_sim_fix is None:
+            map_sim_fix = {}
+        self.map_sim_fix = map_sim_fix
+
+        if scale_map_sim_fix is None:
+            scale_map_sim_fix = {key: LIN for key in map_sim_fix}
+        self.scale_map_sim_fix = scale_map_sim_fix
+
+
+class ParameterMapping(Sequence):
+    """Parameter mapping for multiple conditions.
+
+    This can be used like a list of `ParameterMappingForCondition`s.
+
+    :param parameter_mappings:
+        List of parameter mappings for specific conditions.
+    """
+
+    def __init__(
+            self,
+            parameter_mappings: List[ParameterMappingForCondition] = None
+    ):
+        super().__init__()
+        if parameter_mappings is None:
+            parameter_mappings = []
+        self.parameter_mappings = parameter_mappings
+
+    def __iter__(self):
+        for mapping in self.parameter_mappings:
+            yield mapping
+
+    def __getitem__(self, item):
+        return self.parameter_mappings[item]
+
+    def __len__(self):
+        return len(self.parameter_mappings)
+
+    def append(
+            self,
+            parameter_mapping_for_condition: ParameterMappingForCondition):
+        """Append a condition specific parameter mapping."""
+        self.parameter_mappings.append(parameter_mapping_for_condition)
+
+
+def fill_in_parameters(
+        edatas: List[amici.ExpData],
+        problem_parameters: Dict[str, numbers.Number],
+        scaled_parameters: bool,
+        parameter_mapping: ParameterMapping,
+        amici_model: amici.Model) -> None:
+    """Fill fixed and dynamic parameters into the edatas (in-place).
+
+    :param edatas:
+        List of experimental datas with everything except parameters filled.
```

Comment:
```suggestion
        List of ``ExpData``s with everything except parameters filled.
```
or whatever was proper sphinx for that :grimacing:

Reply:
should be  :class:`amici.amici.ExpData`